url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/Simulation.java

```diff
@@ -0,0 +1,1042 @@
+			byte[] buffer = new byte[1024];
+			FileOutputStream fos = new FileOutputStream(filename + ".zip");
+			ZipOutputStream zos = new ZipOutputStream(fos);
+			ZipEntry ze = new ZipEntry(dbname + ".csv");
+			zos.putNextEntry(ze);
+			FileInputStream in = new FileInputStream(filename);
+			int len;
+			while ((len = in.read(buffer)) > 0) {
+				zos.write(buffer, 0, len);
+			}
+			in.close();
+			zos.closeEntry();
+
+			ze = new ZipEntry(dbname + "_exp_pow.csv");
+			zos.putNextEntry(ze);
+			filename = resources_path + dbname + "_exp_pow.csv";
+			File csvFile2 = new File(filename);
+			in = new FileInputStream(filename);
+			while ((len = in.read(buffer)) > 0) {
+				zos.write(buffer, 0, len);
+			}
+			in.close();
+			zos.closeEntry();
+
+			zos.close();
+			fos.close();
+			csvFile.delete();
+			csvFile2.delete();
+			System.out.println("End of Zipping...");
+			long endTime = System.currentTimeMillis();
+			System.out.println("Time elapsed for Run " + dbname + ": " + ((endTime - startTime)/(1000.0 * 60)) + " mins");
+			System.out.println("Run " + dbname + " ended @ " + Calendar.getInstance().getTime());
+		} catch(Exception e) {
+			e.printStackTrace();
+			System.out.println(Utils.stackTraceToString(e.getStackTrace()));
+		}
+	}
+	
+
+	/**
+	 * Gets the installations.
+	 *
+	 * @return the installations
+	 */
+	public Collection<Installation> getInstallations () {
+		return installations;
+	}
+
+	/**
+	 * Gets the installation at {@code index}.
+	 *
+	 * @param index the index
+	 * @return the installation
+	 */
+	public Installation getInstallation (int index) {
+		return installations.get(index);
+	}
+
+	/**
+	 * Gets the current tick.
+	 *
+	 * @return the current tick
+	 */
+	public int getCurrentTick () {
+		return tick;
+	}
+
+	/**
+	 * Gets the end tick.
+	 *
+	 * @return the end tick
+	 */
+	public int getEndTick () {
+		return endTick;
+	}
+
+	/**
+	 * Gets the random number generator.
+	 *
+	 * @return the orng
+	 */
+	public ORNG getOrng() {
+		return orng;
+	}
+	
+	/**
+	 * Gets the simulation parameters set.
+	 *
+	 * @return the simulation world
+	 */
+	public SimulationParams getSimulationWorld () {
+		return simulationWorld;
+	}
+
+	
+	/**
+	 * Prints the aggregate KPIs, followed by the KPIs for all installations, appliances and activities.
+	 */
+	public void printKPIs()
+	{
+		System.out.println();
+		printKPIs(m.getKPIs(DBResults.AGGR), null, DBResults.AGGR);
+		for (Installation inst: this.getInstallations())
+		{
+			printKPIs(m.getKPIs(inst.getId()), "installation", inst.getId());
+			for (Appliance app: inst.getAppliances())
+				printKPIs(m.getAppKPIs(app.getId()), "appliance", app.getId());
+			for (Person p: inst.getPersons())
+				for (Activity a: p.getActivities())
+					printKPIs(m.getActKPIs(a.getId()), p.getName() + "'s activity", a.getId());
+		}
+	}
+
+	protected void printKPIs( HashMap<String, Double> kpis, String entityType, String entityId)
+	{
+		if (entityType == null && entityId == DBResults.AGGR)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.